scripts/verify-phase-21-production-like-hardening-boundary.sh

```
@@ -0,0 +1,192 @@
+require_file "${phase18_doc}" "Missing Phase 18 live ingest contract doc"
+require_file "${phase17_doc}" "Missing Phase 17 runtime config contract doc"
+require_file "${architecture_doc}" "Missing architecture overview doc"
+require_file "${docs_test}" "Missing Phase 21 docs unittest"
+require_file "${validation_test}" "Missing Phase 21 validation unittest"
+require_file "${workflow_path}" "Missing CI workflow"
+
+roadmap_path="$(find "${repo_root}" -type f -name "${roadmap_filename}" -print -quit)"
+roadmap_relative_path=""
+
+if [[ -n "${roadmap_path}" ]]; then
+  roadmap_relative_path="${roadmap_path#${repo_root}/}"
```

Shellcheck (0.11.0)

[info] Expansions inside ${..} need to be quoted separately, otherwise they match as patterns.

(SC2295)